### PR TITLE
Refactor smoke tests

### PR DESCRIPTION
**Pull request by zancas**
This includes the commit in the other PR.   It is a simplification to the smoke test code.

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -5,8 +5,7 @@ pub mod subcomponents;
 pub mod utils;
 
 use self::subcomponents::{
-    GenerateResponse, GetBlockChainInfoResponse, GetInfoResponse,
-    ZGetNewAddressResponse,
+    GetBlockChainInfoResponse, GetInfoResponse,
```
Comment (AloeareV) [inline src/client.rs:8]: This feels like an unnecessary regression to me. Why are we removing these?

File: tests/zcashrpc-smoke-tests.rs [hunk 2]
```diff
@@ -1,45 +1,22 @@
...
-        }
-    }
-}
-
 macro_rules! run_smoketest {
     ($x:ident) => {
         #[tokio::test]
         async fn $x() {
-            let _res = make_client().$x().await.unwrap();
+            let _res = zcashrpc::client::utils::make_client(true)
```
Comment (AloeareV) [inline tests/zcashrpc-smoke-tests.rs:5]: I strongly approve of this change, very nice.

File: tests/zcashrpc-smoke-tests.rs [hunk 3]
```diff
@@ -1,45 +1,22 @@
...
+                .unwrap();
         }
     };
 }
 
 run_smoketest!(getinfo);
 run_smoketest!(getblockchaininfo);
-run_smoketest!(z_getnewaddress);
+
+#[test]
```
Comment (AloeareV) [inline tests/zcashrpc-smoke-tests.rs:16]: This is neat.

File: src/client/subcomponents.rs [hunk 4]
```diff
@@ -1 +1,96 @@
-zcashrpc_macros::declare_rpc_response_types!();
+//! Sub-components of response messages.
```
Comment (AloeareV) [inline src/client/subcomponents.rs:1]: I appreciate the need for explicitness, but I don't feel like going back to hand rolled types is the way to go, even short term. That said, I suppose it isn't a crime to make use of types that have _already_ been hand_rolled. It still feels regressive to me, though.

File: Cargo.toml [hunk 5]
```diff
@@ -24,6 +24,7 @@ dirs = { version = "*", optional = true }
 proc-macro2 = "1.0"
 quote = "1.0"
 cfg-if = "0.1.10"
+rust_decimal = "*"
```
Comment (AloeareV) [inline Cargo.toml:27]: Oooh! Exciting!

File: src/lib.rs [hunk 6]
```diff
@@ -12,4 +12,4 @@ pub use client::Client;
 pub use error::{Error, ResponseResult};
 
 /// The `ZecAmount` type alias is used to document where ZEC-denominated fields are used. Note that this does not represent Zatoshi-denominated units.
-pub type ZecAmount = f64;
+pub type ZecAmount = rust_decimal::Decimal;
```
Comment (AloeareV) [inline src/lib.rs:15]: I like the look of this, I gotta say.

File: src/client/subcomponents.rs (hunk 4 above)
Comment (zancas) [reply src/client/subcomponents.rs:1]: I cannot read `zcashrpc_macros::declare_rpc_response_types`.

File: src/client.rs (hunk 1 above)
Comment (AloeareV) [reply src/client.rs]: Aaaaand they're back!

**Pull request merged**